### ISSUE TITLE
Citrix DNE Lightweight Filter break DCO adapter

### ISSUE THREAD

**Issue by oyarzun** [OpenVPN/ovpn-dco-win#31]
I upgraded to OpenVPN 2.6 and my connection was failing. Disabling DCO allowed the VPN to connect. After some debugging I discover the DCO adapter was visible in the Device Manager, Control Panel Network Adapters, but not in `ipconfig /all`.

Initially I thought it was because OpenVPN Interactive service was crashing each time I started the VPN, but that seemed to be caused by `block-outside-dns` after it failed to configure the interface. Commenting out `block-outside-dns` stopped the service from crashing.

I ended fixing the adapter not showing up in `ipconfig`, by uninstalling OpenVPN and uninstalling Sonicwall Global VPN Client, reboot, then reinstalling OpenVPN.

I'm not sure this is a bug in the adapter, but figured it might help some else who might be having the same or similar problem.

**Comment by lstipakov** [comment]
Are you saying interactive service crashes with `block-outside-dns` if Sonicwall Global VPN Client is installer?

**Comment by oyarzun** [comment]
@lstipakov yes the interactive service was crashing in the following situation:

1. Sonciwall Global VPN Client installed, but not running.
2. OpenVPN config was attempting to use the DCO adapter
3. `block-outside-dns` was in the client config.
4. The DCO adapter was missing from `ipconfig`.

If I commented out `block-outside-dns` and the other 3 conditions still applied, the service would not crash. But the VPN connection would not work since it never completed setting up the interface.

**Comment by oyarzun** [comment]
I was able to replicate the issue by:

1. uninstalling openvpn
2. installing sonicwall
3. installing openvpn 

Here is what I found in the event viewer regarding the openvpnserv.exe crash:

```
Fault bucket 1481040186744968023, type 4
Event Name: APPCRASH
Response: Not available
Cab Id: 0

Problem signature:
P1: openvpnserv.exe
P2: 2.6.0.5
P3: 63d14fc1
P4: fwpuclnt.dll
P5: 10.0.19041.2486
P6: 871f7448
P7: c0000005
P8: 000000000000a908
P9: 
P10: 

Attached files:
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER8B64.tmp.dmp
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER8BB3.tmp.WERInternalMetadata.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER8BD4.tmp.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER8BD2.tmp.csv
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER8BF2.tmp.txt

These files may be available here:
\\?\C:\ProgramData\Microsoft\Windows\WER\ReportArchive\AppCrash_openvpnserv.exe_bb227d4147eb3611c7436f44e71a93142d21db0_613f2746_dd7091db-bf79-4448-9e8b-f7e266d0a063

Analysis symbol: 
Rechecking for solution: 0
Report Id: ba797b67-0cd2-43a9-8bae-ff03ca1648f5
Report Status: 268435456
Hashed bucket: 278bffa4208dda63048db633f5031357
Cab Guid: 0
```

Here is the log file [ovpn-sonicwall-otc.log](https://github.com/OpenVPN/ovpn-dco-win/files/10548326/ovpn-sonicwall-otc.log)

**Comment by lstipakov** [comment]
@oyarzun Thank you, I was able to reproduce the issue with DCO adapter being semi-functional when Sonicwall client is installed. I will look into that. Meanwhile could you check if this workaround works for you?

 - in Control Panel, go to DCO adapter settings and on the "Network" tab uncheck "DNE LightWeight filter"

**Comment by lstipakov** [comment]
So we have two issues here:

 - compatibility between Sonicwall filter driver and DCO driver, which will be handled in this issue
 - access violation in interactive service, to be handled in [separate](https://github.com/OpenVPN/openvpn/issues/232) issue

**Comment by oyarzun** [comment]
> * in Control Panel, go to DCO adapter settings and on the "Network" tab uncheck "DNE LightWeight filter"

Thanks, I can confirm that this works around the issue.

**Comment by lstipakov** [comment]
Will close when I005 is published.

**Comment by lstipakov** [comment]
Fixed in 2.6.0-I005.
